## Python Module 2 Exercises

1. Print out a data type of a variable?

In [3]:
var1 = 123
type(var1)

int

2. Round 4.5667 to the nearest hundreth using a build-in function, round().

In [5]:
print(round(4.5667, 2))

4.57


3. Convert "657" to an integer data type

In [6]:
int("657")

657

4. Write a function to calculate the miles per gallon.
Get a miles driven and gallons used from a user and call your function to calculate the miles per gallon. 

In [28]:
miles = int(input("Enter miles driven: "))
gallons = int(input("Enter gallons of gas used: "))

def getMilesPerGallon(m,g):
    print(f"Miles Per Gallon: {m/g}")

getMilesPerGallon(miles,gallons)

Enter miles driven: 150
Enter gallons of gas used: 15
Miles Per Gallon: 10.0


5. Enhance your function from exercise 4. Additionally, get a gas price per gallon from a user and write a program to compute the total gas cost and cost per mile. 

The output should look something like this:

    Enter miles driven:            150
    Enter gallons of gas used:     15
    Enter cost per gallon:          3

    Miles Per Gallon:   10.0
    Total Gas Cost:     45.0
    Cost per Mile:      0.3


In [29]:
miles = int(input("Enter miles driven: "))
gallons = int(input("Enter gallons of gas used: "))
costPerGallon = int(input("Enter cost per gallon: "))

print()

def getTotalBreakdown(m,g,c):
    print(f"Miles Per Gallon: {m/g}")
    print(f"Total Gas Cost: {g*c/1}")
    print(f"Cost Per Mile: {c / (m/g)}")
    
getTotalBreakdown(miles,gallons,costPerGallon)


Enter miles driven: 150
Enter gallons of gas used: 15
Enter cost per gallon: 3

Miles Per Gallon: 10.0
Total Gas Cost: 45.0
Cost Per Mile: 0.3


6. Write a function called multiply_two() that accepts two parameters. The function should print a message like '15 + 2 = 17'.  Randomly generate two numbers between 1 and 100, then call the function with the two random numbers.

In [33]:
import random as rand

def multiply_two(num1, num2):
    print(f"{num1} * {num2} = {num1*num2}")

val1 = rand.randint(1,100)
val2 = rand.randint(1,100)

multiply_two(val1,val2)

27 * 66 = 1782


7. Write a function called make_shirt() that accepts a size and the text of a message. The function should print a sentence summarizing the size of the shirt and the message printed on it. The default shirt size is large and default message is 'I love Python!' Make a default size shirt with a message, a medium shirt with the default message, and a shirt of any size with a different message.

In [39]:
def make_shirt(size = "L", msg ="I love Python!"):
    print(f"Shirt size: {size}, Shirt Message: {msg}")
    
make_shirt(msg="I love C#")
make_shirt("M")
make_shirt("XL", "Programming is fun!")

Shirt size: L, Shirt Message: I love C#
Shirt size: M, Shirt Message: I love Python!
Shirt size: XL, Shirt Message: Programming is fun!


8. Create a function for a multiplication table.  Get a number from a user and then call you function which produces a multiplication table for that number.
The output should look something like this:

        Enter a number: 3
        3 X 1 = 3
        3 X 2 = 6
        3 X 3 = 9
        3 X 4 = 12
        3 X 5 = 15
        3 X 6 = 18
        3 X 7 = 21
        3 X 8 = 24
        3 X 9 = 27
        

In [42]:
def generateMultTable(num):
    print(f"{num} * {1} = {num * 1}")
    print(f"{num} * {2} = {num * 2}")
    print(f"{num} * {3} = {num * 3}")
    print(f"{num} * {4} = {num * 4}")
    print(f"{num} * {5} = {num * 5}")
    print(f"{num} * {6} = {num * 6}")
    print(f"{num} * {7} = {num * 7}")
    print(f"{num} * {8} = {num * 8}")
    print(f"{num} * {9} = {num * 9}")

userNum = int(input("Enter a number: "))
generateMultTable(userNum)

Enter a number: 5
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45


9. Write a function that accepts a string and returns a string that is 
    - without any leading nor trailing blanks
    - converts to all capital letters
    - and replaces all blanks with a period(.).  

Write a program to accept a string from a user then call your function and display the original string and changed string.  

The output should look something like this:

        Enter a text: may your dreams come true
        
        Original Text: may your dreams come true
        New Text: MAY.YOUR.DREAMS.COME.TRUE

In [45]:
def transformStr(text):
    print(f"Original Text: {text}")
    print(f"New Text: {text.strip().upper().replace(' ', '.')}")

userStr = input("Enter text: ")
print()
transformStr(userStr)

Enter text: This is a string of characters

Original Text: This is a string of characters
New Text: THIS.IS.A.STRING.OF.CHARACTERS


10. Write a function that returns a randomly selected character from a passed string. Write a program to accept a string from a user then call your function and display the original string and selected character.

The output should look something like this:

        Enter a text: functions
        Original Text:	functions
        Random character:	s

In [46]:
import random as rand

def chooseRandCharacter(text):
    return text[rand.randint(0,len(userStr)-1)]

userStr = input("Enter text: ")
print(f"Original Text: {userStr}")
print(f"Random character: {chooseRandCharacter(userStr)}")

Enter text: This is a string of data
Original Text: This is a string of data
Random character: a


11. Write a program that asks the user to enter the width and length of a room. Once the values have been read, your program should compute and display the area of the room. The length and the width will be entered as floating point numbers. Include units in your prompt and output message; either feet or meters, depending on which unit you are more comfortable working with.

In [50]:
length = float(input("Enter the length (in meters): "))
width = float(input("Enter the width (in meters): "))

print(f"The area of the space is: {length*width} m-squared")

Enter the length (in meters): 5.8
Enter the width (in meters): 22.7
The area of the space is: 131.66 m-squared


12. The program that you create for this exercise will begin by reading the cost of a meal ordered at a restaurant from the user. Then your program will compute the tax and tip for the meal. Use your local tax rate when computing the amount of tax owing. Compute the tip as 18 percent of the meal amount (without the tax). The output from your program should include the tax amount, the tip amount, and the grand total for the meal including both the tax and the tip. Format the output so that all of the values are displayed using two decimal places.

In [49]:
cost = float(input("Enter the cost of the meal: "))
tax = cost*.07
tip = cost*.18
grandTotal = cost + tax + tip

print()
print(f"Subtotal: ${cost}")
print(f"Tax: ${tax:,.2f}")
print(f"Tip: ${tip:,.2f}")
print(f"Grand Total: ${grandTotal:,.2f}")

Enter the cost of the meal: 5.29

Subtotal: $5.29
Tax: $0.37
Tip: $0.95
Grand Total: $6.61


13. Write a program that reads a positive integer, n, from the user and then displays the sum of all of the integers from 1 to n. The sum of the first n positive integers can be computed using the formula: sum = (n)(n+1)/2

In [10]:
userNum = int(input("Enter a number: "))
print(f"The sum of the numbers from 1 to {userNum} is: ", userNum*(userNum+1)//2)

Enter a number: 10
The sum of the numbers from 1 to 10 is:  55
